client/src/components/LibraryFolder/TopToolbar/FolderTopBar.vue

```diff
@@ -0,0 +1,382 @@
+import mod_path_bar from "components/LibraryFolder/path-bar";
+import mod_import_dataset from "./import-to-history/import-dataset";
+import mod_import_collection from "./import-to-history/import-collection";
+import mod_add_datasets from "./add-datasets";
+import { Toast } from "ui/toast";
+import download from "./download";
+import mod_utils from "utils/utils";
+import { getAppRoot } from "onload/loadConfig";
+
+initTopBarIcons();
+
+Vue.use(BootstrapVue);
+
+export default {
+    name: "FolderTopBar",
+    props: {
+        folder_id: {
+            type: String,
+            required: true,
+        },
+        include_deleted: {
+            type: Boolean,
+            required: true,
+        },
+        selected: {
+            type: Array,
+            required: true,
+        },
+        metadata: {
+            type: Object,
+            required: true,
+        },
+        folderContents: {
+            type: Array,
+            required: true,
+        },
+    },
+    components: {
+        FontAwesomeIcon,
+    },
+    data() {
+        return {
+            dataset_manipulation: false,
+            logged_dataset_manipulation: false,
+            is_admin: false,
+            multiple_add_dataset_options: false,
+            user_library_import_dir: false,
+            library_import_dir: false,
+            allow_library_path_paste: false,
+            list_genomes: [],
+            list_extensions: [],
+            // datatype placeholder for extension auto-detection
+            auto: {
+                id: "auto",
+                text: "Auto-detect",
+                description:
+                    "This system will try to detect the file type automatically." +
+                    " If your file is not detected properly as one of the known formats," +
+                    " it most likely means that it has some format problems (e.g., different" +
+                    " number of columns on different rows). You can still coerce the system" +
+                    " to set your data to the format you think it should be." +
+                    " You can also upload compressed files, which will automatically be decompressed.",
+            },
+        };
+    },
+    created() {
+        const Galaxy = getGalaxyInstance();
+        this.is_admin = Galaxy.user.attributes.is_admin;
+        this.user_library_import_dir = Galaxy.config.user_library_import_dir;
+        this.library_import_dir = Galaxy.config.library_import_dir;
+        this.allow_library_path_paste = Galaxy.config.allow_library_path_paste;
+        if (
+            this.user_library_import_dir !== null ||
+            this.allow_library_path_paste !== false ||
+            this.library_import_dir !== null
+        ) {
+            this.multiple_add_dataset_options = true;
+        }
+        const contains_file_or_folder = this.folderContents.find((el) => el.type === "folder" || el.type === "file");
```

Comment:
`===` is usually only required when comparing against `0` in order to avoid implicit conversion. In these cases `==` is sufficient.

Reply:
`==` will work, but I can't think of a good reason not to be explicit and require type be a string?